extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php

```diff
@@ -29,7 +29,7 @@ private function getCategoryRequestUrl() {
 		return self::DISCUSSIONS_API_BASE_DEV . "$this->cityId/forums?responseGroup=small&viewableOnly=true";
 	}
 
-	public function getCategoryId( $category ) {
+	public function getCategoryName( $categoryId ) {
 		$memcKey = wfMemcKey( __METHOD__, self::MCACHE_VER );
 		$rawData = WikiaDataAccess::cache(
```

Comment:
I noticed that the raw response is cached, and then searched every time `categoryNameLookup` is called.  What about only caching the info needed?  Something like:
```php
	function() {
		$rawData = $this->apiRequest( $this->getCategoryRequestUrl() );

		$validForums = [];
		if ( empty( $rawData['_embedded']['doc:forum'] ) ) {
			return $validForums;
		}

		$categories = $rawData['_embedded']['doc:forum'];
		foreach ( $categories as $value ) {
			$id = $value['id'];
			$name = $value['name'];

			$validForums[$id] = $name;
		}
		return $validForums;
	}
```
Then line 42 could be simplified to:
```php
return empty( $data[$categoryId] ) ? false : $data[$categoryId]
```
I think having this prepared as an array will also make building multi-category URLs a lot easier too.